docs/fsharp/style-guide/conventions.md

```diff
@@ -687,6 +687,50 @@ module Array =
 
 For legacy reasons some string functions in FSharp.Core still treat nulls as empty strings and do not fail on null arguments. However do not take this as guidance, and do not adopt coding patterns that attribute any semantic meaning to "null".
 
+### Leverage F# 9 null syntax at the API boundaries
+
+F# 9 adds [syntax](../language-reference/values/null-values.md#null-values-starting-with-f-9) to explicitly state that a value can be null. It's designed to be used on the API boundaries, to make the compiler indicate the places where null handling null is missing.
+
+Here is an example of the valid usage of the syntax:
+
+```fsharp
+let processStream (stream: System.IO.StreamReader) =
```

Comment:
The example has some value, but it is unrelated to the text:
The text speaks about API boundaries, but here the `processLine` is private to the function, so not exposed to the outside world...

Reply:
Yeah, fair point. I updated the example but still don't like it too much... it might help to add something from the real world, like implementing some tiny .NET interface with nulls - would you have something from the top of your head?

Reply:
.Equals() or CompareTo() are classical null-flooded examples of a public API :)

Reply:
Right - good idea.